core/src/test/java/org/bitcoinj/core/TransactionTest.java

```diff
@@ -641,4 +641,19 @@ public void getWeightAndVsize() {
         assertEquals(542, tx.getWeight());
         assertEquals(136, tx.getVsize());
     }
+
+    @Test
+    public void edgeCases() {
+        MessageSerializer serializer = UNITTEST.getDefaultSerializer();
+        // Non SegWit tx with zero input and outputs
+        String txHex = "01000000000000000000";
+        Transaction tx = serializer.makeTransaction(HEX.decode(txHex));
+        assertEquals(txHex, HEX.encode(tx.bitcoinSerialize()));
+
```

Comment:
Can you split this into two test cases which can fail individually? And maybe a bit more speaking names?